packages/client/utils/getAnonymousId.ts

```diff
@@ -6,6 +6,8 @@ const retrieveGtagClientId = async (
 ): Promise<string | undefined> => {
   return new Promise((resolve) => {
     gtag('get', measurementId, 'client_id', resolve)
+    // if the call gets blocked, then it might never resolve
+    window.setTimeout(() => resolve(undefined), 1000)
```

Comment:
@tianrunhe just tagging you so you can see there is a timeout now.

Reply:
Yeah I've taken a look at your PR yesterday. Thanks for fixing this!